packages/mui-material/src/utils/memoTheme.ts

```diff
@@ -0,0 +1,31 @@
+import { CSSInterpolation } from '@mui/styled-engine';
+import { Theme } from '../styles/createTheme';
+
+type ThemeStyleFunction = (props: { theme: Theme }) => CSSInterpolation;
+
+// We need to pass an argument as `{ theme }` for PigmentCSS, but we don't want to
+// allocate more objects.
+const arg = { theme: undefined as unknown as Theme };
+
+/**
+ * Memoize style function on theme.
+ * Intended to be used in styled() calls that only need access to the theme.
+ */
+export default function memoTheme(styleFn: ThemeStyleFunction) {
```

Comment:
The location of the helper doesn't seem to make sense from a product perspective, issue open #43440.